geomstats/geometry/dirichlet_distributions.py

```diff
@@ -0,0 +1,322 @@
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n_samples]
+            Sample from beta distributions.
+        """
+        geomstats.errors.check_belongs(point, self)
+        point = gs.to_ndarray(point, to_ndim=2)
+        samples = []
+        for i in range(n_samples):
+            samples.append(gs.array(
+                dirichlet.rvs(point[i, :], size=n_samples)))
+        return samples[0] if len(point) == 1 else gs.stack(samples)
+
+
+class DirichletMetric(RiemannianMetric):
+    """Class for the Fisher information metric on Dirichlet distributions."""
+
+    def __init__(self, dim):
+        super(DirichletMetric, self).__init__(dim=dim)
+
+    def metric_matrix(self, base_point=None):
+        """Compute the inner-product matrix.
+
+        Compute the inner-product matrix of the Fisher information metric
+        at the tangent space at base point.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        mat : array-like, shape=[..., dim, dim]
+            Inner-product matrix.
+        """
+        if base_point is None:
+            raise ValueError('A base point must be given to compute the '
+                             'metric matrix')
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        n_points = base_point.shape[0]
+
+        mat_ones = gs.ones((n_points, self.dim, self.dim))
+        poly_sum = gs.polygamma(1, gs.sum(base_point, -1))
+        mat_diag = gs.zeros((n_points, self.dim, self.dim))
+        diag = gs.einsum('ijj->ij', mat_diag)
+        diag[:] = gs.polygamma(1, base_point)
+
+        mat = mat_diag - gs.einsum('i,ijk->ijk', poly_sum, mat_ones)
+        return gs.squeeze(mat)
+
+    def christoffels(self, base_point):
+        """Compute the Christoffel symbols.
+
+        Compute the Christoffel symbols of the Fisher information metric.
+
+        Parameters
+        ----------
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        christoffels : array-like, shape=[..., dim, dim, dim]
+            Christoffel symbols.
+        """
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
+        n_points = base_point.shape[0]
+
+        def coefficients(ind_k):
+            param_k = base_point[..., ind_k]
+            param_sum = gs.sum(base_point, -1)
+            C1 = 1 / gs.polygamma(1, param_k) / (
+                1 / gs.polygamma(1, param_sum)
+                - gs.sum(1 / gs.polygamma(1, base_point), -1))
+            C2 = - C1 * gs.polygamma(2, param_sum) / gs.polygamma(1, param_sum)
+
+            mat_ones = gs.ones((n_points, self.dim, self.dim))
+            mat_diag = gs.zeros((n_points, self.dim, self.dim))
+            diag = gs.einsum('ijj->ij', mat_diag)
+            diag[:] = - gs.polygamma(2, base_point) \
+                / gs.polygamma(1, base_point)
```

Comment:
Are mat_diag and diag used?

Reply:
If the point is to modify `mat_diag` by filling `diag`, you could also use `geomstats.algebra_utils.from_vector_to_diagonal_matrix(diag_vector)` instead.

Reply:
Did not know this function existed, nice !